<!DOCTYPE html><html lang="en"><head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"><meta content="yes" name="apple-mobile-web-app-capable"><meta content="black-translucent" name="apple-mobile-web-app-status-bar-style"><meta content="telephone=no" name="format-detection"><meta name="description" content=""><title> | 本店今天不营业</title><link rel="stylesheet" type="text/css" href="/css/style.css?v=1.0.0"><link rel="stylesheet" type="text/css" href="https://unpkg.com/normalize.css"><link rel="stylesheet" type="text/css" href="https://unpkg.com/purecss/build/pure-min.css"><link rel="stylesheet" type="text/css" href="https://unpkg.com/purecss/build/grids-responsive-min.css"><link rel="stylesheet" href="https://unpkg.com/font-awesome@4.7.0/css/font-awesome.min.css"><script type="text/javascript" src="https://unpkg.com/jquery/dist/jquery.min.js"></script><link rel="icon" mask="" sizes="any" href="/favicon.ico"><link rel="Shortcut Icon" type="image/x-icon" href="/favicon.ico"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="apple-touch-icon-precomposed" href="/apple-touch-icon.png"><script type="text/javascript" src="https://unpkg.com/clipboard/dist/clipboard.min.js"></script><script type="text/javascript" src="https://unpkg.com/toastr/build/toastr.min.js"></script><link rel="stylesheet" href="https://unpkg.com/toastr/build/toastr.min.css"><meta name="generator" content="Hexo 7.3.0"></head><script>MathJax = {
  tex: {
    inlineMath: [['$', '$'], ['\\(', '\\)']],  // Enable \( ... \) and $ ... $
    displayMath: [['$$', '$$'], ['\\[', '\\]']]
  },
  options: {
    skipHtmlTags: ['script', 'noscript', 'style', 'textarea', 'pre'], // Avoid processing unwanted tags
    processHtmlClass: 'mathjax-process'  // Apply MathJax only to elements with this class
  }
};</script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script><body><div class="body_container"><div id="header"><div class="site-name"><h1 class="hidden">本店今天不营业</h1><a id="logo" href="/.">本店今天不营业</a><p class="description"></p></div><div id="nav-menu"><a class="current" href="/."><i class="fa fa-home"> Home</i></a><a href="/archives/"><i class="fa fa-archive"> Archive</i></a><a href="/about/"><i class="fa fa-user"> About</i></a><a href="/atom.xml"><i class="fa fa-rss"> RSS</i></a></div></div><div class="pure-g" id="layout"><div class="pure-u-1 pure-u-md-3-4"><div class="content_container"><div class="post"><h1 class="post-title"></h1><div class="post-meta">2025-08-07</div><div class="post-content"><p>哈哈，都快做完了才发现压根用不到，只需要他们搞得raw protein data，相思了<br>哦哦，其实也还好，簇也用得上<br>主要是我太傻逼了，挪数据到硬盘时候忘记把原始数据导进来了，后面在硬盘上弄的时候直接把原始数据的事给忘了。。。</p>
<h1 id="Data"><a href="#Data" class="headerlink" title="Data"></a>Data</h1><p><a target="_blank" rel="noopener" href="https://portal.nersc.gov/MGV/">https://portal.nersc.gov/MGV/</a></p>
<p>referencing from <a target="_blank" rel="noopener" href="https://github.com/snayfach/MGV">https://github.com/snayfach/MGV</a>:<br>Nayfach et al. Metagenomic compendium of 189,680 DNA viruses from the human gut microbiome. 2021. <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41564-021-00928-6">https://www.nature.com/articles/s41564-021-00928-6</a>.</p>
<h2 id="setup"><a href="#setup" class="headerlink" title="setup"></a>setup</h2><h3 id="1-Viral-detection-pipeline"><a href="#1-Viral-detection-pipeline" class="headerlink" title="1. Viral detection pipeline"></a>1. <a target="_blank" rel="noopener" href="https://github.com/snayfach/MGV/blob/master/viral_detection_pipeline/README.md">Viral detection pipeline</a></h3><ul>
<li><a target="_blank" rel="noopener" href="https://github.com/hyattpd/Prodigal">Prodigal v2.6.3</a><ul>
<li><pre><code>conda install bioconda::prodigal
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">  - or</span><br><span class="line">    https://github.com/hyattpd/Prodigal/wiki/installation#installing-on-mac-os-x</span><br><span class="line"></span><br><span class="line">- HMMER v3.1b2</span><br><span class="line">  - trying [HMMER v3.4](https://github.com/EddyRivasLab/hmmer) (latest version)</span><br></pre></td></tr></table></figure>
conda install bioconda::hmmer
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">    - ~~sadly, v3.4 works differently than v3.1b2, I&#x27;ll go back to v3.1b2 for now.. but~~</span><br><span class="line">      - ~~haha, this is why I love cs, v3.1b2 doesn&#x27;t even compile~~</span><br><span class="line">      - ~~going back to v3.4~~</span><br><span class="line">      - ~~actually I **accidentally** skipped over the `prodigal` step so v3.4 didn&#x27;t work at the first time~~</span><br><span class="line">  - [v3.1b2](http://hmmer.org/download.html)</span><br><span class="line"></span><br><span class="line">- VirFinder v1.1</span><br><span class="line">  - glmnet  `install.packages(&quot;glmnet&quot;, dependencies=TRUE)`</span><br><span class="line">  - Rcpp    `install.packages(&quot;Rcpp&quot;, dependencies=TRUE)`</span><br><span class="line">  - [qvalue](https://github.com/StoreyLab/qvalue)</span><br><span class="line">    - I tried:</span><br></pre></td></tr></table></figure>
  install.packages(&quot;devtools&quot;)
  library(&quot;devtools&quot;)
  install_github(&quot;jdstorey/qvalue&quot;)
  <figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">  - then [VirFinder](https://github.com/jessieren/VirFinder?tab=readme-ov-file#installation)</span><br><span class="line">    - But the code provided didn&#x27;t work for me.</span><br><span class="line">    - The file we downloaded technically won&#x27;t work **directly**.</span><br><span class="line">    - We will only need a few step from there, depress the downloaded `zip` or `.tar.gz` file, then go to the specific folder that matches your OS, `install.package` the VirFinder in that folder.</span><br><span class="line"></span><br><span class="line">### 2. Quality Control</span><br><span class="line">- [CheckV](https://bitbucket.org/berkeleylab/checkv/src/master/)</span><br><span class="line">  - Some users may wish to update the database using their own complete genomes:</span><br></pre></td></tr></table></figure>
checkv update_database /path/to/checkv-db /path/to/updated-checkv-db genomes.fna
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">- There are two ways to run CheckV:</span><br><span class="line">  - Using a single command to run the full pipeline (recommended):</span><br></pre></td></tr></table></figure>
  checkv end_to_end input_file.fna output_directory -t 16
  <figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">- Using individual commands for each step in the pipeline:</span><br></pre></td></tr></table></figure>
  checkv contamination input_file.fna output_directory -t 16
  checkv completeness input_file.fna output_directory -t 16
  checkv complete_genomes input_file.fna output_directory
  checkv quality_summary input_file.fna output_directory
  <figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">  - Please refer to Berkeley Lab Checkv for specific functions it has, or meaning of output.</span><br><span class="line"> </span><br><span class="line">- Installing</span><br><span class="line">  - conda install didn&#x27;t work for me,need a few additional package</span><br><span class="line">    - rust</span><br></pre></td></tr></table></figure>
  conda install rust
  <figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">- diamond (use version 2.1.8 simply because a bug showed up for 2.1.9, and the newest version 2.1.12 is also not compatible, so went for the guaranteed version)</span><br></pre></td></tr></table></figure>
  conda install bioconda::diamond=2.1.8
  <figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">### 3. Cluster genomes based on ANI</span><br></pre></td></tr></table></figure>
conda install bioconda::blast
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">- other than this just follow along, I need this because I am missing it, you may even skip this installation</span><br><span class="line"></span><br><span class="line">### 4. Cluster genomes based on AAI</span><br></pre></td></tr></table></figure>
conda install bioconda::mcl
</code></pre>
</li>
</ul>
<pre><code>-  Stijn van Dongen, Graph Clustering Via a Discrete Uncoupling Process,
  SIAM Journal on Matrix Analysis and Applications, 30(1):121-141, 2008.
  ( http://link.aip.org/link/?SJMAEL/30/121/1 )
</code></pre>
</li>
<li>same, just follow along the manual if you are not missing anything.</li>
</ul>
<h3 id="5-Create-SNP-phylogenetic-trees"><a href="#5-Create-SNP-phylogenetic-trees" class="headerlink" title="5. Create SNP phylogenetic trees"></a>5. Create SNP phylogenetic trees</h3><ul>
<li>Package<ul>
<li>MUMmer v4.0.0beta2</li>
<li>FastTreeMP v2.1.10</li>
<li>newer versions are fine</li>
</ul>
</li>
</ul>
<h3 id="6-marker-tree"><a href="#6-marker-tree" class="headerlink" title="6. marker tree"></a>6. marker tree</h3><ul>
<li><p>It was at this moment that I knew, protein data is all that I need, maybe some of the previous clusters, oh yeah, the clusters are helpful, but what’s really needed, also for the later on foldseek part is that raw protein data.</p>
</li>
<li><p>Pipeline</p>
<ul>
<li>So notice that, by following the previous steps we can’t necessarily get several representative fna files to create a snp tree</li>
</ul>
</li>
</ul>
<h1 id="step"><a href="#step" class="headerlink" title="step"></a>step</h1></div><div class="tags"></div><div class="post-nav"><a class="next" href="/2024/09/11/Basic-Algorithms/">Basic-Algorithms</a></div></div></div></div><div class="pure-u-1-4 hidden_mid_and_down"><div id="sidebar"><div class="widget"><form class="search-form" action="//www.google.com/search" method="get" accept-charset="utf-8" target="_blank"><input type="text" name="q" maxlength="20" placeholder="Search"/><input type="hidden" name="sitesearch" value="http://example.com"/></form></div><div class="widget"><div class="author-info"><a class="info-avatar" href="/about/" title="About"><img class="nofancybox" src="/img/avatar.png"/></a><p>To be a better man.</p><a class="info-icon" href="https://twitter.com/username" title="Twitter" target="_blank" style="margin-inline:5px"> <i class="fa fa-twitter-square" style="margin-inline:5px"></i></a><a class="info-icon" href="mailto:admin@domain.com" title="Email" target="_blank" style="margin-inline:5px"> <i class="fa fa-envelope-square" style="margin-inline:5px"></i></a><a class="info-icon" href="https://github.com/username" title="Github" target="_blank" style="margin-inline:5px"> <i class="fa fa-github-square" style="margin-inline:5px"></i></a><a class="info-icon" href="/atom.xml" title="RSS" target="_blank" style="margin-inline:5px"> <i class="fa fa-rss-square" style="margin-inline:5px"></i></a></div></div><div class="widget"><div class="widget-title"><i class="fa fa-folder-o"> Categories</i></div></div><div class="widget"><div class="widget-title"><i class="fa fa-star-o"> Tags</i></div><div class="tagcloud"></div></div><div class="widget"><div class="widget-title"><i class="fa fa-file-o"> Recent</i></div><ul class="post-list"><li class="post-list-item"><a class="post-list-link" href="/2025/08/07/MGV-Connector/">MGV-Connector</a></li><li class="post-list-item"><a class="post-list-link" href="/2024/09/11/Basic-Algorithms/">Basic-Algorithms</a></li><li class="post-list-item"><a class="post-list-link" href="/2024/09/05/Cuttag-Tutorial-Tutorial/">Cuttag Tutorial Tutorial</a></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-external-link"> Links</i></div><ul></ul><a href="http://www.example1.com/" title="site-name1" target="_blank">site-name1</a><ul></ul><a href="http://www.example2.com/" title="site-name2" target="_blank">site-name2</a><ul></ul><a href="http://www.example3.com/" title="site-name3" target="_blank">site-name3</a></div></div></div><div class="pure-u-1 pure-u-md-3-4"><div id="footer">Copyright © 2025 <a href="/." rel="nofollow">本店今天不营业.</a> Powered by<a rel="nofollow" target="_blank" href="https://hexo.io"> Hexo.</a><a rel="nofollow" target="_blank" href="https://github.com/tufu9441/maupassant-hexo"> Theme</a> by<a rel="nofollow" target="_blank" href="https://github.com/pagecho"> Cho.</a></div></div></div><a class="show" id="rocket" href="#top"></a><script type="text/javascript" src="/js/totop.js?v=1.0.0" async></script><script type="text/javascript" src="https://unpkg.com/@fancyapps/fancybox/dist/jquery.fancybox.min.js" async></script><script type="text/javascript" src="/js/fancybox.js?v=1.0.0" async></script><link rel="stylesheet" type="text/css" href="https://unpkg.com/@fancyapps/fancybox/dist/jquery.fancybox.min.css"><script type="text/javascript" src="/js/copycode.js?v=1.0.0" successtext="Copy Successed!"></script><link rel="stylesheet" type="text/css" href="/css/copycode.css?v=1.0.0"><script type="text/javascript" src="/js/codeblock-resizer.js?v=1.0.0"></script><script type="text/javascript" src="/js/smartresize.js?v=1.0.0"></script></div></body></html>